src/anm/analytics.js

```diff
@@ -0,0 +1,77 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics');
+
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
+            } else {
+                var trackUrl = animatronUrl + '?player=' + encodeURIComponent(data);
+                sendViaGif(trackUrl);
+            }
+        } else {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (trackUrl) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = trackUrl;
+        beacon.onerror = beacon.onload = function () {
+            beacon.onerror = beacon.onload = null;
+            setTimeout(event, timeout);
+        }
+    };
+    event();
```

Comment:
It's "premature optimization" in your case, since it will never send any data, why call it? I'm just against of this particular call, since it does nothing useful at all. No need to "complicate code".